src/auto/functions.rs

```diff
@@ -721,86 +726,81 @@ pub fn listenv() -> Vec<std::ffi::OsString> {
-//    unsafe { TODO: call glib_sys:g_log_default_handler() }
-//}
-
 pub fn log_remove_handler(log_domain: &str, handler_id: u32) {
     unsafe {
         glib_sys::g_log_remove_handler(log_domain.to_glib_none().0, handler_id);
     }
 }
 
-//pub fn log_set_always_fatal(fatal_mask: /*Ignored*/LogLevelFlags) -> /*Ignored*/LogLevelFlags {
-//    unsafe { TODO: call glib_sys:g_log_set_always_fatal() }
-//}
-
-//pub fn log_set_default_handler(log_func: /*Unimplemented*/Fn(&str, /*Ignored*/LogLevelFlags, &str), user_data: /*Unimplemented*/Option<Fundamental: Pointer>) -> /*Unimplemented*/Fn(&str, /*Ignored*/LogLevelFlags, &str) {
-//    unsafe { TODO: call glib_sys:g_log_set_default_handler() }
-//}
-
-//pub fn log_set_fatal_mask(log_domain: &str, fatal_mask: /*Ignored*/LogLevelFlags) -> /*Ignored*/LogLevelFlags {
-//    unsafe { TODO: call glib_sys:g_log_set_fatal_mask() }
-//}
-
-//pub fn log_set_handler(log_domain: Option<&str>, log_levels: /*Ignored*/LogLevelFlags, log_func: /*Unimplemented*/Fn(&str, /*Ignored*/LogLevelFlags, &str), user_data: /*Unimplemented*/Option<Fundamental: Pointer>) -> u32 {
-//    unsafe { TODO: call glib_sys:g_log_set_handler() }
-//}
-
-//#[cfg(any(feature = "v2_46", feature = "dox"))]
-//pub fn log_set_handler_full(log_domain: Option<&str>, log_levels: /*Ignored*/LogLevelFlags, log_func: /*Unimplemented*/Fn(&str, /*Ignored*/LogLevelFlags, &str), user_data: /*Unimplemented*/Option<Fundamental: Pointer>) -> u32 {
-//    unsafe { TODO: call glib_sys:g_log_set_handler_full() }
-//}
-
-//#[cfg(any(feature = "v2_50", feature = "dox"))]
-//pub fn log_set_writer_func(func: /*Unimplemented*/Fn(/*Ignored*/LogLevelFlags, /*Ignored*/Vec<LogField>, usize) -> /*Ignored*/LogWriterOutput, user_data: /*Unimplemented*/Option<Fundamental: Pointer>) {
-//    unsafe { TODO: call glib_sys:g_log_set_writer_func() }
-//}
+pub fn log_set_always_fatal(fatal_mask: LogLevelFlags) -> LogLevelFlags {
+    unsafe { from_glib(glib_sys::g_log_set_always_fatal(fatal_mask.to_glib())) }
+}
 
-//#[cfg(any(feature = "v2_50", feature = "dox"))]
-//pub fn log_structured(log_domain: &str, log_level: /*Ignored*/LogLevelFlags, : /*Unknown conversion*//*Unimplemented*/Fundamental: VarArgs) {
-//    unsafe { TODO: call glib_sys:g_log_structured() }
-//}
+pub fn log_set_fatal_mask(log_domain: &str, fatal_mask: LogLevelFlags) -> LogLevelFlags {
+    unsafe {
+        from_glib(glib_sys::g_log_set_fatal_mask(
+            log_domain.to_glib_none().0,
+            fatal_mask.to_glib(),
+        ))
+    }
+}
 
-//#[cfg(any(feature = "v2_50", feature = "dox"))]
-//pub fn log_structured_array(log_level: /*Ignored*/LogLevelFlags, fields: /*Ignored*/&[&LogField]) {
-//    unsafe { TODO: call glib_sys:g_log_structured_array() }
-//}
+#[cfg(any(feature = "v2_50", feature = "dox"))]
+pub fn log_structured_array(log_level: LogLevelFlags, fields: &[&LogField]) {
```

Comment:
@GuillaumeGomez You were not listening to me :P Don't bind the structured logging functions in this round, they require much more effort to get right.